tool/log.re

```diff
@@ -32,21 +31,19 @@ fn clean_expired_messages() {
 }
 
 fn draw_log_messages() {
-    let mut i = 0;
-    while i < LOG.messages.len() {
-        let viewport = Tas::get_viewport_size();
-        let message = LOG.messages.get(i).unwrap();
-        Tas::draw_text(DrawText {
-            text: message.content,
-            color: Color { red: 1., green: 0., blue: 0., alpha: 1.},
-            x: 10.,
-            // 51 denotes the amount of vertical space from the bottom of the screen.
-            // 48 denotes the amount of vertical space between each log message.
-            y: viewport.height.to_float() - ((51. * SETTINGS.ui_scale) + ((48. * SETTINGS.ui_scale) * i.to_float())),
-            scale: SETTINGS.ui_scale,
-            scale_position: false,
-        });
-        i += 1;
-    }:
+    let viewport = Tas::get_viewport_size();
+    let mut messages = "";
+    for message in LOG.messages {
+        messages = f"{message.content}\n{messages}";
+    }
+    let txt_size = Tas::get_text_size(messages, SETTINGS.ui_scale);
```

Comment:
If you add an `e`, you make the code 20% more readable by making the variable only 12% longer ;)